src/PowerShellEditorServices/Services/DebugAdapter/DebugService.cs

```diff
@@ -687,8 +688,20 @@ private async Task FetchGlobalAndScriptVariablesAsync()
             var scopeVariableContainer = new VariableContainerDetails(this.nextVariableId++, "Scope: " + scope);
             this.variables.Add(scopeVariableContainer);
 
-            IReadOnlyList<PSObject> results = await _executionService.ExecutePSCommandAsync<PSObject>(psCommand, CancellationToken.None)
-                .ConfigureAwait(false);
+            IReadOnlyList<PSObject> results;
+            try
+            {
+                results = await _executionService.ExecutePSCommandAsync<PSObject>(psCommand, CancellationToken.None)
+                    .ConfigureAwait(false);
+            }
+            catch (CmdletInvocationException ex)
+            {
+                if (!ex.ErrorRecord.CategoryInfo.Reason.Equals("PSArgumentOutOfRangeException"))
+                {
+                    throw;
```

Comment:
I figured that would be handled by the higher order catch, but we can handle it here too.

Reply:
I'm gonna leave it as-is, because it represents a situation we don't know about and don't know how to handle, so this should bubble up to the user at a higher order level, but it shouldn't crash the program, it should display a vscode error window and then end the variable handling.